plugins/HHAnalyzer.cc

```diff
@@ -801,6 +577,9 @@ void HHAnalyzer::analyze(const edm::Event& event, const edm::EventSetup&, const
             myllmetjj.DPhi_llmet_jj = fabs(ROOT::Math::VectorUtil::DeltaPhi(llmet[illmet].p4, jj[ijj].p4));
             myllmetjj.cosThetaStar_CS = fabs(getCosThetaStar_CS(llmet[illmet].p4, jj[ijj].p4));
             myllmetjj.MT_fullsystem = myllmetjj.p4.Mt();
+            myllmetjj.melaAngles = getMELAAngles(llmet[illmet].p4, jj[ijj].p4, leptons[ilep1].p4, leptons[ilep2].p4, jets[ijet1].p4, jets[ijet2].p4);
+            myllmetjj.visMelaAngles = getMELAAngles(ll[ill].p4, jj[ijj].p4, leptons[ilep1].p4, leptons[ilep2].p4, jets[ijet1].p4, jets[ijet2].p4); // only take the visible part of the H(ww) candidate
+
             // Some selection
             if (myllmetjj.minDR_l_j < m_minDR_l_j_Cut)
```

Comment:
IMO this is the way to go. First, you clean the jets from the lepton, and then you do what you have to do :+1: